app/assets/javascripts/hubstats/application.js

```diff
@@ -14,45 +14,72 @@
 //= require jquery_ujs
 //= require select2
 //= require hubstats/bootstrap
+//= require bootstrap-datepicker
 //= require_tree .
 
 $(document).ready( function() {
-  setDateRange()
+  $('.input-daterange').datepicker({
+    "todayHighlight": true,
+    "endDate": "Today",
+    "todayBtn": true
+  });
+  setDateRange();
 });
 
 function setDateRange() {
-  var index = readCookie("hubstats_index") || 2;
-  var timer = document.getElementById("time-select");
+  var cookie = readCookie("hubstats_dates");
 
-  timer.selectedIndex = index;
+  if (cookie === null || cookie.indexOf("null") > -1) {
+    var dates = getDefaultDateRange();
```

Comment:
For cases where you can't assign them at the top of the function, you should still declare them at the top of the function.
```
var dates;
...
dates = ...
```